### PR TITLE
Add tests for AnyValue size edge cases in dynamic dispatch

### PR DESCRIPTION
**Pull request by jvepsalainen-nv**
## Summary

- Test empty struct (zero data bytes) conforming to an interface, verifying AnyValue packing handles zero-payload types alongside types with data.
- Verify that explicit `[anyValueSize(4)]` produces error 41011 when a conforming type requires 8 bytes.
- Test both exact-fit and oversized explicit `[anyValueSize]`, verifying pack/unpack produces correct dispatch results in both cases.

## Test plan

- [x] `slang-test tests/language-feature/dynamic-dispatch/anyvalue-zero-size-impl.slang` passes (2 values verified)
- [x] `slang-test tests/language-feature/dynamic-dispatch/anyvalue-explicit-too-small.slang` passes (diagnostic verified)
- [x] `slang-test tests/language-feature/dynamic-dispatch/anyvalue-explicit-sizing.slang` passes (4 values verified)

### CODE REVIEW

File: tests/language-feature/dynamic-dispatch/anyvalue-explicit-sizing.slang [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    int a;
+    int b;
+    int getValue() { return a + b; }
+}
+
+[anyValueSize(32)]
+interface IOversized
+{
+    int getValue();
+}
```
[the remark this answers is not in the document]
Comment (jvepsalainen-nv) [reply tests/language-feature/dynamic-dispatch/anyvalue-explicit-sizing.slang:35]: Fixed in f472ad6. Added IR dump check via -dump-ir-after inferAnyValueSizeWhereNecessary that verifies AnyValueSize(8) and AnyValueSize(32) are preserved in the IR for IExact and IOversized respectively.

File: tests/language-feature/dynamic-dispatch/anyvalue-zero-size-impl.slang [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+interface IFoo
+{
+    int getValue();
+}
+
+struct EmptyImpl : IFoo
+{
+    int getValue() { return 42; }
+}
```
[the remark this answers is not in the document]
Comment (jvepsalainen-nv) [reply tests/language-feature/dynamic-dispatch/anyvalue-zero-size-impl.slang:19]: Fixed in f472ad6. Added IR dump check that verifies AnyValueSize(4), EmptyImpl with SizeAndAlignment 0 bytes, and DataImpl with SizeAndAlignment 4 bytes.

File: tests/language-feature/dynamic-dispatch/anyvalue-explicit-too-small.slang [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// Verify that an explicit [anyValueSize(4)] produces an error when a
+// conforming type requires more than 4 bytes of storage. LargeImpl has
+// two int fields (8 bytes total), which exceeds the 4-byte limit.
```
Comment (skiminki-nv) [inline tests/language-feature/dynamic-dispatch/anyvalue-explicit-too-small.slang]: Usually, it makes sense to test the first failing value over the boundary (BVEC-style). This is effective in testing off-by-ones such as wrong comparisons (e.g., `<` instead if `<=`) in implementation.

Comment (jvepsalainen-nv) [reply tests/language-feature/dynamic-dispatch/anyvalue-explicit-too-small.slang]: Fixed in eff084f. Changed \[anyValueSize(4)]\ to \[anyValueSize(7)]\ so the test exercises the exact boundary (7 < 8) rather than a value far below the required size.
